jfix-stdlib-socket/src/main/java/ru/fix/stdlib/socket/proxy/ProxySocket.java

```diff
@@ -0,0 +1,91 @@
+package ru.fix.stdlib.socket.proxy;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.ServerSocket;
+import java.net.Socket;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+public class ProxySocket implements AutoCloseable {
+
+    private static Logger log = LoggerFactory.getLogger(ProxySocket.class);
+
+    private String destinationHost;
+    private int destinationPort;
+    private int sourcePort;
+
+    private ServerSocket sourceServerSocket;
+    private AtomicBoolean isShutdown = new AtomicBoolean();
+
+    private ExecutorService executorService = Executors.newFixedThreadPool(25);
```

Comment:
It will be better to use provided executor service and not hardcode particular implementation with fixed numbers of thread.   What if user decided to proxy 26 connections and not 25?  

Be aware that usually we are using NamedExecutors.newThreadPool* - they are profiled. 

It will be nice to provide documentation that each connection to the socket will use separate thread from the executorService.